app/admin/RTMediaUploadTermsAdmin.php

```diff
@@ -0,0 +1,166 @@
+			'args' => array(
+				'key' => 'activity_enable_upload_terms',
+				'value' => $options['activity_enable_upload_terms'],
+				'desc' => __( 'User have to check the terms and conditions before uploading the media.', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+		$render_options['general_upload_terms_page_link'] = array(
+			'title' => __( 'Link for "Terms of Service" page', 'buddypress-media' ),
+			'callback' => array( 'RTMediaFormHandler', 'textbox' ),
+			'args' => array(
+				'key' => 'general_upload_terms_page_link',
+				'value' => $options['general_upload_terms_page_link'],
+				'desc' => __( 'Link to the terms and condition page where user can read terms and conditions.', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+		// add extra field for admin setting
+		$render_options['general_upload_terms_message'] = array(
+			'title' => __( 'Terms of Service Message', 'buddypress-media' ),
+			'callback' => array( 'RTMediaFormHandler', 'textbox' ),
+			'args' => array(
+				'key' => 'general_upload_terms_message',
+				'value' => isset( $options['general_upload_terms_message'] ) ? $options['general_upload_terms_message'] : $this->upload_terms_message,
+				'desc' => __( 'Add Terms of Service Message.', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+		$render_options['general_upload_terms_error_message'] = array(
+			'title' => __( 'Error Message', 'buddypress-media' ),
+			'callback' => array( 'RTMediaFormHandler', 'textbox' ),
+			'args' => array(
+				'key' => 'general_upload_terms_error_message',
+				'value' => isset( $options['general_upload_terms_error_message'] ) ? $options['general_upload_terms_error_message'] : $this->upload_terms_error_message,
+				'desc' => __( 'Display Error Message When User Upload Media Without selecting checkbox .', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+		$render_options['general_upload_terms_show_pricacy_message'] = array(
+			'title' => __( 'Show "Privacy Message" on website', 'buddypress-media' ),
+			'callback' => array( 'RTMediaFormHandler', 'checkbox' ),
+			'args' => array(
+				'key' => 'general_upload_terms_show_pricacy_message',
+				'value' => $options['general_upload_terms_show_pricacy_message'],
+				'desc' => __( 'User will see the privacy message on website.', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+		$render_options['general_upload_terms_privacy_message'] = array(
+			'title' => __( 'Privacy Message', 'buddypress-media' ),
+			'callback' => array( 'RTMediaFormHandler', 'textarea' ),
+			'args' => array(
+				'key' => 'general_upload_terms_privacy_message',
+				'value' => isset( $options['general_upload_terms_privacy_message'] ) ? $options['general_upload_terms_privacy_message'] : '',
+				'desc' => __( 'Display privacy message on your website.', 'buddypress-media' ),
+			),
+			'group' => 40,
+		);
+
+		return $render_options;
+	}
+}
+
+// Instantiate object
+new RTMediaUploadTermsAdmin();
```

Comment:
Run PHPCS on your changes with standard `WordPress` in `app/admin/RTMediaUploadTermsAdmin.php` file. @malavasita